src/caches/basic_method_caches.jl

```diff
@@ -155,15 +159,18 @@ function alg_cache(alg::RKMil_General,prob,u,ΔW,ΔZ,p,rate_prototype,noise_rate
   L = zero(noise_rate_prototype)
   mil_correction = zero(u)
   ggprime = zero(noise_rate_prototype)
-  m = (typeof(ΔW) <: Number || is_diagonal_noise(prob))*1 + (alg.is_commutative)*2 + (!(typeof(ΔW) <: Number || is_diagonal_noise(prob) || alg.is_commutative))*3
-  WikJ = fill_WikJ(ΔW,Val(m),Val(true))
-  #
-  # if typeof(ΔW) <: Number || is_diagonal_noise(prob)
-  #   WikJ = WikJDiagonal_iip(ΔW,Val(1))
-  # elseif alg.is_commutative
-  #   WikJ = WikJCommute_iip(ΔW,Val(2))
-  # else
-  #   WikJ = WikJGeneral_iip(ΔW,Val(3))
-  # end
+  # m = (typeof(ΔW) <: Number || is_diagonal_noise(prob))*1 + (alg.is_commutative)*2 + (!(typeof(ΔW) <: Number || is_diagonal_noise(prob) || alg.is_commutative))*3
+  # WikJ = fill_WikJ(ΔW,Val(m),Val(true))
+
+  if typeof(ΔW) <: Number || is_diagonal_noise(prob)
+    # WikJ = WikJDiagonal_iip(ΔW)
+    WikJ = fill_WikJ(ΔW,Val(1),Val(true))
```

Comment:
?

Reply:
I still don't understand the purpose of Value types here. 
Earlier what I did was
```
  m = (typeof(ΔW) <: Number || is_diagonal_noise(prob))*1 + (alg.is_commutative)*2 + (!(typeof(ΔW) <: Number || is_diagonal_noise(prob) || alg.is_commutative))*3
  WikJ = fill_WikJ(ΔW,Val(m),Val(false))
```
Got compilation error.
Then I changed it to:
```
    WikJ = WikJDiagonal_oop(ΔW,Val(true))
```
But I guess this won't make any difference because there is only 1 definition of `WikJDiagonal_oop`

So I did
```
    WikJ = fill_WikJ(ΔW,Val(1),Val(false))
```
1, 2, or 3 for Diagonal, commutative and general noise and true or false for in place and out of place.

Reply:
no, the purpose was for `is_commutative` to be something that is a value type so it can be a compile-time check. Examples: http://www.stochasticlifestyle.com/type-dispatch-design-post-object-oriented-programming-julia/

Reply:
But then we'll have to take it as an argument. Like we can default it to diagonal noise case but then for general noise, the user will have to give another argument.